### PR TITLE
Add X25519MLKEM768

### PR DESCRIPTION
**Pull request by bwesterb**

### CODE REVIEW

File: draft-kwiatkowski-tls-ecdhe-mlkem.md [hunk 1]
```diff
@@ -46,78 +59,97 @@ informative:
          ANSI: ANS X9.62-2005
 --- abstract
 
-This draft defines a hybrid key agreement for TLS 1.3 that combines
+This draft defines X25519MLKEM768 and SecP256r1MLKEM768 two hybrid key agreements for TLS 1.3 that combines
```
Comment (chris-wood) [inline draft-kwiatkowski-tls-ecdhe-mlkem.md:62]:
```suggestion
This draft defines X25519MLKEM768 and SecP256r1MLKEM768, two hybrid key agreements for TLS 1.3 that combines
```

File: draft-kwiatkowski-tls-ecdhe-mlkem.md [hunk 2]
```diff
@@ -46,78 +59,97 @@ informative:
...
-scheme using elliptic curves from ANSI X9.62 [ECDSA] and NIST SP 800-186
-{{?DSS=DOI.10.6028/NIST.SP.800-186}}.
+This document introduces two new supported groups for hybrid post-quantum key
+agreements in TLS 1.3: X25519MLKEM768 and SecP256r1MLKEM768.
+Both combine ML-KEM-768 with ECDH in the manner of {{hybrid}}.
+The first uses X25519 {{rfc7748}} as elliptic curve to match X25519Kyber768Draft00.
+The second uses secp256r1 (NIST P-256) {{ECDSA}} {{?DSS=DOI.10.6028/NIST.SP.800-186}}.
 
-The new group enables the derivation of TLS session keys using FIPS-approved schemes. NIST's
+Both groups enables the derivation of TLS session keys using FIPS-approved schemes. NIST's
```
Comment (chris-wood) [inline draft-kwiatkowski-tls-ecdhe-mlkem.md:87]:
```suggestion
Both groups enable the derivation of TLS session keys using FIPS-approved schemes. NIST's
```

File: draft-kwiatkowski-tls-ecdhe-mlkem.md [hunk 3]
```diff
@@ -46,78 +59,97 @@ informative:
...
+When the X25519MLKEM768 group is negotiated, the client's key_exchange value
+is the concatenation of the client's ML-KEM-768 encapsulation key
+and the client's X25519 ephemeral share.
+The size of the client share is 1216 bytes (1184 bytes for the ML-KEM part and 32 bytes for X25519.)
 
-When this group is negotiated, the client's share is a fixed-size concatenation of
-the ECDHE share and ML-KEM's public key. The ECDHE share is the serialized value of
+When the SecP256r1MLKEM768 group is negotiated, the client's key_exchange value
+is the concatenation of the secp256r1 ephemeral share and ML-KEM-768 encapsulation key.
+The ECDHE share is the serialized value is the serialized value of
```
Comment (chris-wood) [inline draft-kwiatkowski-tls-ecdhe-mlkem.md:106]:
```suggestion
The ECDHE share is the serialized value of
```

File: draft-kwiatkowski-tls-ecdhe-mlkem.md [hunk 4]
```diff
@@ -134,12 +166,35 @@ This document requests/registers a new entry to the TLS Supported Groups
  : This document
 
  Comment:
- : Combining secp256r1 ECDH with the ML-KEM-768
+ : Combining secp256r1 ECDH with ML-KEM-768
+
+## X25519MLKEM768
+
+ Value:
+ : 25500 (0x639C)
```
Comment (chris-wood) [inline draft-kwiatkowski-tls-ecdhe-mlkem.md:174]:
```suggestion
 : 25500 (0x639C)
```

didn’t we discuss 0x11ec?

Comment (bwesterb) [reply draft-kwiatkowski-tls-ecdhe-mlkem.md:174]: First time I heard that one.

Comment (jschanck) [reply draft-kwiatkowski-tls-ecdhe-mlkem.md:174]: I floated 0x11ec  it on Signal. It looks like "0x concat ec". I'm not attached to it.

Comment (bwesterb) [reply draft-kwiatkowski-tls-ecdhe-mlkem.md:174]: Ah! Sorry, missed that. I like it.

**Comment by kriskwiatkowski** [comment]
Perfect! Thanks

**Pull request merged**